lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -114,8 +114,19 @@ export class PublicClientApplication {
         const cachedHash = this.browserStorage.getItem(TemporaryCacheKeys.URL_HASH);
         try {
             // If hash exists, handle in window. Otherwise, cancel any current requests and continue.
-            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.config.auth.navigateToLoginRequestUrl);
-            const responseHash = UrlString.hashContainsKnownProperties(hash) ? hash : cachedHash;
+            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage);
+            let responseHash: string;
+            if (this.config.auth.navigateToLoginRequestUrl && UrlString.hashContainsKnownProperties(hash)) {
+                this.browserStorage.setItem(TemporaryCacheKeys.URL_HASH, hash);
+                interactionHandler.navigateToRequestUrl();
+                return;
+            } else if (!this.config.auth.navigateToLoginRequestUrl) {
+                responseHash = UrlString.hashContainsKnownProperties(hash) ? hash : cachedHash;
+                window.location.hash = "";
+            } else {
+                responseHash = cachedHash;
+            }
```

Comment:
```suggestion
            const validHash = UrlString.hashContainsKnownProperties(hash);

            if (this.config.auth.navigateToLoginRequestUrl && validHash) {
            this.browserStorage.setItem(TemporaryCacheKeys.URL_HASH, hash);
            const loginRequestUrl = this.browserStorage.getItem(TemporaryCacheKeys.ORIGIN_URI);
            interactionHandler.navigateToRequestUrl(loginRequestUrl);
            return;
            }

            if (!this.config.auth.navigateToLoginRequestUrl) {
            window.location.hash = "";
            }

            const responseHash = validHash ? hash : cachedHash;
```

Reply:
Are we back porting (I believe only the syntax) to `msal-core`?

Reply:
I've done something similar to this - please take a look